pkg/ipcache/ipcache.go

```diff
@@ -525,6 +526,26 @@ func (ipc *IPCache) RemoveMetadata(prefix netip.Prefix, resource ipcacheTypes.Re
 	ipc.TriggerLabelInjection()
 }
 
+func (ipc *IPCache) UpsertPrefixes(prefixes []netip.Prefix, src source.Source, resource ipcacheTypes.ResourceID) {
```

Comment:
```
diff --git a/pkg/ipcache/ipcache.go b/pkg/ipcache/ipcache.go
index a23d935185c5..fdc29b3129a3 100644
--- a/pkg/ipcache/ipcache.go
+++ b/pkg/ipcache/ipcache.go
@@ -526,6 +526,12 @@ func (ipc *IPCache) RemoveMetadata(prefix netip.Prefix, resource ipcacheTypes.Re
        ipc.TriggerLabelInjection()
 }
 
+// UpsertPrefixes inserts the prefixes into the IPCache and associates CIDR
+// labels with these prefixes, thereby making these prefixes selectable in
+// policy via local ("CIDR") identities.
+//
+// This will trigger asynchronous calculation of any datapath updates necessary
+// to implement the logic associated with the new CIDR labels.
 func (ipc *IPCache) UpsertPrefixes(prefixes []netip.Prefix, src source.Source, resource ipcacheTypes.ResourceID) {
        ipc.metadata.Lock()
        for _, p := range prefixes {
@@ -536,6 +542,17 @@ func (ipc *IPCache) UpsertPrefixes(prefixes []netip.Prefix, src source.Source, r
        ipc.TriggerLabelInjection()
 }
 
+// RemovePrefixes removes the association between the prefixes and the CIDR
+// labels corresponding to those prefixes.
+//
+// This is the reverse operation of UpsertPrefixes(). If multiple callers call
+// UpsertPrefixes() with different resources, then RemovePrefixes() will only
+// remove the association for the target resource. That is, *all* callers must
+// call RemovePrefixes() before this the these prefixes become disassociated
+// from the "CIDR" labels.
+//
+// This will trigger asynchronous calculation of any datapath updates necessary
+// to implement the logic associated with the removed CIDR labels.
 func (ipc *IPCache) RemovePrefixes(prefixes []netip.Prefix, src source.Source, resource ipcacheTypes.ResourceID) {
        ipc.metadata.Lock()
        for _, p := range prefixes {
```